plugin/proxy/grpc_test.go

```diff
@@ -59,6 +64,97 @@ func TestStartupShutdown(t *testing.T) {
 	}
 }
 
+func TestRunAQuery(t *testing.T) {
+	grpclog.SetLogger(discard{})
+
+	upstream := &staticUpstream{
+		from: ".",
+		HealthCheck: healthcheck.HealthCheck{
+			Hosts: pool(),
+		},
+	}
+	g := newGrpcClient(nil, upstream)
+	upstream.ex = g
+
+	p := &Proxy{}
+	p.Upstreams = &[]Upstream{upstream}
+
+	err := g.OnStartup(p)
+	if err != nil {
+		t.Errorf("Error starting grpc client exchanger: %s", err)
+		return
```

Comment:
`return` is not needed, `Fatalf` is you want to quit the test.

Reply:
ok. Will also change for the others cases in same file